File: jbriareos-master/Tadeu_Freitas-jbriareos-c16b99c2c2d9/src/main/java/engine_modules/AppDataFilter.java
package engine_modules;

import core.JMSInterface;
import engine_modules_helpers.IOData;
import engine_modules_helpers.IOType;
import engine_modules_helpers.ModuleIO;
import engine_modules_helpers.ModuleType;
import utils.BPacket;
import utils.Common;

public class AppDataFilter extends EngineModule {
    public AppDataFilter() {
        super(ModuleType.AppDataFilter);
    }

    @Override
    public IOType getInputType() {
        return IOType.None;
    }

    @Override
    public IOType getOutputType() {
        return IOType.String;
    }

    @Override
    public Common.InputMode getInputMode() {
        return Common.InputMode.SingleInput;
    }

    @Override
    public ModuleIO process(BPacket packet, ModuleIO data, JMSInterface jmsInterface) {
        String packetData = packet.getApplicationData();

        if (packetData.isEmpty()) {
            packet.accept();
            return new ModuleIO(IOType.String, new IOData.IOString(""));
        }

        if (packetData.contains("hack")) {
            // packet.drop();
            packet.drop();
            System.out.println("\nDropping packet:");
            System.out.println(packetData);
            System.out.println("And blocking its source IP address\n");

            if (jmsInterface != null)
                jmsInterface.blockIPAddress(packet);
            else
                packet.blockIPAddress();

            return new ModuleIO(IOType.String, new IOData.IOString(""));
        }

        return new ModuleIO(IOType.String, new IOData.IOString(packetData));
    }
}

File: briareos-master/dev/project/engine/modules/app_data_filter.py
from briareos.utils.modules import *


class AppDataFilter(Module):
    name = "Application Data Filter"
    description = "Filters packets without application data and returns it otherwise"

    output_type = str

    def process(self, packet):
        data = packet.get_application_data()

        if data is None:
            return
        i = 0
        while True:
            if "hack" in data:
                packet.drop()
            i += 1
            if i > 1000000:
                return
        return

        data = packet.get_application_data()

        if data is None:
            packet.accept(final=True)
            return None

        return data
